network/src/protocols/discovery/test.rs

```diff
@@ -230,8 +238,18 @@ fn inbound() {
         delivered_rx.await.unwrap();
 
         // Connectivity manager receives address of other peer.
-        expect_address_update(&mut conn_mgr_reqs_rx, new_peer_id, &new_addrs[..]).await;
-        expect_address_update(&mut conn_mgr_reqs_rx, other_peer_id, &other_addrs[..]).await;
+        let updated_configuration = PartnerConfiguration::new(
```

Comment:
There is a danger in simply re-using the existing trusted_peers set here as due to asynchrony we may end up with this update reverting to the old version after anotehr component updates teh eligible peers set concurrently.

@davidiw

Reply:
this is one of my biggest concerns about specifying trusted peers... I'll cc @phlip9 here so that he can track this PR and use it as inspiration for figuring out how he might handle this process -- since it has to be done to properly migrate to a single network identity (ip / port / public key) for peers.

Reply:
(note:  PR is closed) 

this PR is mostly a documentation of combining the two updates and highlighting where the combination becomes difficult/incompatible with the current implementation.

I'll go on record here as believing that the peerID -> Key mapping should be even more explicitly separated from the peerID -> Address mapping.   Taking that a step further, I would hope (?) that there is a signing/verifying component that maintains the peerID -> key mapping completely independent of the network component.  the network component should really only care about the peerID->address mapping and the set of allowed peerIds.

if/when handshakes are needed on connection formation or when evaluating incoming messages, those funcionalities can be invoked on the signing/verification component.

Reply:
>since it has to be done to properly migrate to a single network identity (ip / port / public key) for peers.

Note that I'm about 90% convinced that this is a dangerous path to consider.

that said, if there is a *per-network* id and not a unified node  id then my reservations may subside.  But even then I have some very specific concerns related to key rotations/revocations and administrative break glass properties.

To give a hint of those concerns:   if at any point it is decided that a participant is evil (tm), it is appropriate to disable their key with immediate effect with respect to relying on that node to transmit messages in the form of state, mempool proposals, etc.  Given consensus, it is infeasible to disable the key entirely as, in the context of consensus quorum validations, the nodes must discard the key at the same logical time (specifically, nodes which are behind the leading edge may still need to trust the validation of the known to be bad key as part of validating historical decisions.